alonzo/impl/src/Cardano/Ledger/Alonzo/Tx.hs

```diff
@@ -412,8 +315,27 @@ isTwoPhaseScriptAddress tx addr =
         Just scr -> not (isNativeScript @era scr)
 
 -- | txsize computes the length of the serialised bytes
-instance HasField "txsize" (ValidatedTx era) Integer where
-  getField (ValidatedTxConstr (Memo _ bytes)) = fromIntegral (SBS.length bytes)
+instance
+  ( Typeable era,
+    ToCBOR (Core.TxBody era),
+    ToCBOR (Core.AuxiliaryData era)
+  ) =>
+  HasField "txsize" (ValidatedTx era) Integer
+  where
+  getField tx = fromIntegral (BS.length $ serialize' tx)
+
+instance
+  ( Typeable era,
+    ToCBOR (Core.TxBody era),
+    ToCBOR (Core.AuxiliaryData era)
+  ) =>
+  ToCBOR (ValidatedTx era)
+  where
+  toCBOR (ValidatedTx {body, wits, auxiliaryData}) =
```

Comment:
As an alternative, we could go even further and drop the `ToCBOR` instance entirely, providing a raw function `toCBORForSizeComputation`. The point of this would just be to prevent accidental foot shooting later.

Reply:
Yes that would be better